src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -111,4 +128,109 @@ private void addListingToBusinesses(PlacesSearchResult place) {
+        PlacesSearchResult[] similarBusinessesInTheArea = request
+                                                            .await()
+                                                            .results;
+        if (similarBusinessesInTheArea.length > 1){
+          checkBusinessThroughLinkedin(currentBusiness.getName(), similarBusinessesInTheArea);
+        }
+      }
+    } catch(GeneralSecurityException e) {
+      LOGGER.warning(e.getMessage());
+    } catch(IOException e) {
+      LOGGER.warning(e.getMessage());
+    } catch(ApiException e) {
+      LOGGER.warning(e.getMessage());
+    } catch(InterruptedException e) {
+      LOGGER.warning(e.getMessage());
+    } 
+  }
+  
+  private void checkBusinessThroughLinkedin(String currentBusinessName, 
+                          PlacesSearchResult[] similarBusinessesInTheArea) 
+                              throws GeneralSecurityException, IOException {
+    String cx = "REDACTED"; 
+    Customsearch cs = new Customsearch.Builder(
+                                GoogleNetHttpTransport.newTrustedTransport(), 
+                                JacksonFactory.getDefaultInstance(), 
+                                null) 
+                                .setApplicationName("linkedinSearch") 
+                                .setGoogleClientRequestInitializer(
+                                      new CustomsearchRequestInitializer(KEY)) 
+                                .build();
+
+    Customsearch.Cse.List list = cs.cse().list(currentBusinessName).setCx(cx); 
+    List<Result> searchJsonResults = list.execute().getItems();
+    String[] numberOfFollowers;
+    int companyFollowers = 0;
+    if (searchJsonResults!=null && searchJsonResults.size() != 0) {
+      Result linkedinBusiness = searchJsonResults.get(0);
+      String businessDescription = (String) linkedinBusiness.getPagemap()
+                                                            .get("metatags")
+                                                            .get(0)
+                                                            .get("og:description");
+      if(businessDescription.indexOf(startIndex) != -1 
+          && businessDescription.indexOf(endIndex) != -1){
+        String followers = businessDescription.substring(
+                                businessDescription.indexOf(startIndex) + 2, 
+                                businessDescription.indexOf(endIndex) - 1);
+        companyFollowers = 
+                    Integer.parseInt(followers.replaceAll(",", ""));
+      }
+      if (companyFollowers > minFollowers) {
+        addBigBusinessToDatabase();
+      } else {
+        checkNumberOfSimilarBusinessesInTheArea(currentBusiness.getName(), similarBusinessesInTheArea);
+      }
+    }
+  }
+
+  private void checkNumberOfSimilarBusinessesInTheArea(String businessName, 
+                          PlacesSearchResult[] similarBusinessesInTheArea) {
+    int countNumberOfMatchingBusiness = 0;
+    int i = 0;
```

Comment:
What is the goal here?

Reply:
Well the i variable allows me to iterate through the results from the textSearchRequest, and countNumberOfMatchingBusiness allows me to check how many businesses within the results have the same corresponding name and different addresses.